<launch>
    <!-- This fix the library linkage issue in RoboStack -->
    <env name="LD_PRELOAD" value="$(env CONDA_PREFIX)/lib/libyaml-cpp.so" />

    <arg name="output"            default="screen"    doc="display output to screen or log file"/>
    <arg name="odom_topic"   default="/sim_pose"    doc="ROS topic for the pose"/>
    <arg name="control_topic"   default="/control"    doc="ROS topic for control input"/>
    <arg name="pub_rate"     default="30"        doc="rate at which to publish the pose"/>
    <arg name="init_x"       default="0"        doc="initial x position"/>
    <arg name="init_y"       default="0"        doc="initial y position"/>
    <arg name="init_yaw"       default="0"        doc="initial z position"/>
    
    <node pkg="racecar_interface" type="simulation_node.py" name="simulation_node" output="$(arg output)">
        <param name="odom_topic"           value="$(arg odom_topic)" />
        <param name="control_topic"        value="$(arg control_topic)" />
        <param name="pub_rate"          value="$(arg pub_rate)" />
        <param name="init_x"            value="$(arg init_x)" />
        <param name="init_y"            value="$(arg init_y)" />
        <param name="init_yaw"          value="$(arg init_yaw)" />
    </node>

    <node pkg="racecar_interface" type="visualization_node.py" name="visualization_node" output="$(arg output)">
        <param name="odom_topic"           value="$(arg odom_topic)" />
        <param name="control_topic"        value="$(arg control_topic)"/>
    </node>

    <include file="$(find traj_planning_ros)/launch/ilqr_planning.launch" >
        <arg name="odom_topic" value="$(arg odom_topic)" />
        <arg name="control_topic" value="$(arg control_topic)" />
    </include>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find racecar_interface)/rviz/config.rviz" />
    <node type="rqt_gui" name="rqt_gui" pkg="rqt_gui" args="--perspective-file $(find racecar_interface)/rviz/rqt.perspective"/>
</launch>
